lib/views/home/home_view.dart

```diff
@@ -10,11 +10,43 @@ import 'widgets/search/search_widget.dart';
 import 'widgets/meditation_cards/card_list_widget.dart';
 import 'package:Medito/providers/providers.dart';
 
-class HomeView extends ConsumerWidget {
-  const HomeView({super.key});
+class HomeView extends ConsumerStatefulWidget {
+  HomeView({super.key});
 
   @override
-  Widget build(BuildContext context, WidgetRef ref) {
+  ConsumerState<HomeView> createState() => _HomeViewState();
+}
+
+class _HomeViewState extends ConsumerState<HomeView> {
+  final ScrollController _scrollController = ScrollController();
+  Color _backgroundColor = ColorConstants.ebony;
+  var _announcementColor;
+  var hasAnnouncement = false;
+  @override
+  void initState() {
+    super.initState();
+    _scrollController.addListener(_onScroll);
+  }
+
+  @override
+  void dispose() {
+    _scrollController.dispose();
+    super.dispose();
+  }
+
+  void _onScroll() {
+    final newColor = _scrollController.offset > 100
```

Comment:
Magic number https://hireamir.com/blog/software-anti-patterns-magic-numbers
Please make a note to fix this :) I will merge for now though